tests/e2e/utils/infrastructure.py

```diff
@@ -0,0 +1,197 @@
+import io
+import os
+import sys
+import zipfile
+from abc import ABC, abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Tuple, Type
+
+import boto3
+import yaml
+from aws_cdk import App, AssetStaging, BundlingOptions, CfnOutput, DockerImage, RemovalPolicy, Stack, aws_logs
+from aws_cdk.aws_lambda import Code, Function, LayerVersion, Runtime, Tracing
+
+PYTHON_RUNTIME_VERSION = f"V{''.join(map(str, sys.version_info[:2]))}"
+
+
+class PythonVersion(Enum):
+    V37 = {"runtime": Runtime.PYTHON_3_7, "image": Runtime.PYTHON_3_7.bundling_image.image}
+    V38 = {"runtime": Runtime.PYTHON_3_8, "image": Runtime.PYTHON_3_8.bundling_image.image}
+    V39 = {"runtime": Runtime.PYTHON_3_9, "image": Runtime.PYTHON_3_9.bundling_image.image}
+
+
+class InfrastructureStackInterface(ABC):
+    @abstractmethod
+    def synthesize() -> Tuple[dict, str]:
+        ...
+
+    @abstractmethod
+    def __call__() -> Tuple[dict, str]:
+        ...
+
+
+class InfrastructureStack(InfrastructureStackInterface):
+    def __init__(self, handlers_dir: str, stack_name: str, config: dict) -> None:
+        self.stack_name = stack_name
+        self.handlers_dir = handlers_dir
+        self.config = config
+
+    def _create_layer(self, stack: Stack):
```

Comment:
Does it worth having this in the `Base` class? assuming it'll be universally used for potentially new Infrastructure Stack strategies.

I couldn't quite grasp the extensibility yet, dependency inversion, naming, etc, but let's not delay this PR any further. We can dedicate focus on this in the future, as there's lots of changes already. 

This will be easier to spot as we try to create more tests, more infra, etc.

Reply:
ok!